payroll/tests/test_hr_payroll_cancel.py

```diff
@@ -10,6 +10,12 @@
 
 class TestHrPayrollCancel(common.TransactionCase):
     def setUp(self):
+
+        # Set system parameter
+        self.env["ir.config_parameter"].sudo().set_param(
+            "payroll.allow_cancel_payslips", True
+        )
+
         super(TestHrPayrollCancel, self).setUp()
```

Comment:
This super() should be the first thing you call. That's why the test is failing.